### PR TITLE
[Compute] Add new parameter `community-gallery-info` and `--permission Community` options to `az sig create`, add new command `az sig share enable-community`

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
---

Migrate community gallery from private package to CLI extension https://github.com/Azure/azure-cli/issues/20262
closed https://github.com/Azure/azure-cli/issues/20262

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/image-gallery/azext_image_gallery/custom.py [hunk 1]
```diff
@@ -60,3 +65,75 @@ def list_generator(pages, num_results=50):
...
+            from .vendored_sdks.azure_mgmt_compute.models._models_py3 import SharingProfile
+            gallery.sharing_profile = SharingProfile(permissions=permissions)
+        else:
+            gallery.sharing_profile.permissions = permissions
+    if soft_delete is not None:
+        gallery.soft_delete_policy.is_soft_delete_enabled = soft_delete
+
+    client = _compute_client_factory(cmd.cli_ctx)
+
+    return client.galleries.begin_create_or_update(resource_group_name, gallery_name, gallery, **kwargs)
```
Comment (zhoxing-ms) [inline src/image-gallery/azext_image_gallery/custom.py:139]: Do we need these two methods? If not, please delete them

File: src/image-gallery/setup.py [hunk 2]
```diff
@@ -44,7 +44,7 @@
     name='image_gallery',
     version=VERSION,
     description='Support for Azure Image Gallery',
-   # TODO: Update author and email, if applicable
+    # TODO: Update author and email, if applicable
```
Comment (zhoxing-ms) [inline src/image-gallery/setup.py:47]: Why modify here?

**Comment by zhoxing-ms** [comment]
Please write the description of changes into `HISTORY,rst` and update `setup.py` to release the new extension version

File: src/image-gallery/setup.py (hunk 2 above)
Comment (PARADISSEEKR) [reply src/image-gallery/setup.py:47]: `azdev style` will prompt that it is not aligned

Comment (zhoxing-ms) [reply src/image-gallery/setup.py:47]: OK

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**